be/src/exec/iceberg/iceberg_delete_builder.cpp

```diff
@@ -175,6 +180,66 @@ Status ORCEqualityDeleteBuilder::build(const std::string& timezone, const std::s
     return Status::OK();
 }
 
+Status ParquetEqualityDeleteBuilder::build(const std::string& timezone, const std::string& delete_file_path,
+                                           int64_t file_length,
+                                           const std::shared_ptr<DefaultMORProcessor> mor_processor,
+                                           std::vector<SlotDescriptor*> slot_descs,
+                                           TupleDescriptor* delete_column_tuple_desc,
+                                           const TIcebergSchema* iceberg_equal_delete_schema,
+                                           RuntimeState* state) {
+    std::unique_ptr<RandomAccessFile> file;
+    ASSIGN_OR_RETURN(file, _fs->new_random_access_file(delete_file_path));
+
+    std::unique_ptr<parquet::FileReader> reader;
+    try {
+        reader = std::make_unique<parquet::FileReader>(state->chunk_size(), file.get(), file->get_size().value(), 0);
+    } catch (std::exception& e) {
+        const auto s =
+                strings::Substitute("ParquetEqualityDeleteBuilder::build create parquet::FileReader failed. reason = $0", e.what());
+        LOG(WARNING) << s;
+        return Status::InternalError(s);
+    }
+
+    auto scanner_ctx = std::make_unique<HdfsScannerContext>();
+    auto scan_stats = std::make_unique<HdfsScanStats>();
+    std::vector<HdfsScannerContext::ColumnInfo> columns;
+    THdfsScanRange scan_range;
+    scan_range.offset = 0;
+    scan_range.length = file_length;
+    for (size_t i = 0; i < slot_descs.size(); i++) {
+        auto* slot = slot_descs[i];
+        HdfsScannerContext::ColumnInfo column;
+        column.slot_desc = slot;
+        column.col_idx = i;
+        column.col_type = slot->type();
+        column.slot_id = slot->id();
+        column.col_name = slot->col_name();
+        column.decode_needed = true;
+        columns.emplace_back(std::move(column));
+    }
+    scanner_ctx->timezone = timezone;
+    scanner_ctx->stats = scan_stats.get();
+    scanner_ctx->tuple_desc = delete_column_tuple_desc;
+    scanner_ctx->iceberg_schema = iceberg_equal_delete_schema;
+    scanner_ctx->materialized_columns = columns;
+    scanner_ctx->scan_range = &scan_range;
+    scanner_ctx->lazy_column_coalesce_counter = new std::atomic<int32_t>(0);
+    RETURN_IF_ERROR(reader->init(scanner_ctx.get()));
+
+    while (true) {
+        ChunkPtr chunk = ChunkHelper::new_chunk(*delete_column_tuple_desc, state->chunk_size());
+        Status status = reader->get_next(&chunk);
+        if (status.is_end_of_file()) {
+            break;
+        }
+
+        RETURN_IF_ERROR(status);
+        ChunkPtr& result = chunk;
+        RETURN_IF_ERROR(mor_processor->append_chunk_to_hashtable(result));
```

Comment:
how this append_chunk_to_hashtable is implemented? does it come from hash joiner?